src/Uno.UI/UI/Xaml/UIElement.skia.cs

```diff
@@ -371,6 +377,31 @@ internal virtual void OnArrangeVisual(Rect rect, Rect? clip)
 
 		Visual IVisualElement2.GetVisualInternal() => ElementCompositionPreview.GetElementVisual(this);
 
+		private void Cleanup()
+		{
+			NativeDispatcher.Main.Enqueue(() =>
+			{
+				if (this.GetParent() is UIElement originalParent)
+				{
+					originalParent.RemoveChild(this);
```

Comment:
Is this change related to the leak I found on the Dopes sample?

Reply:
I'm also wondering, if `this` is being GC'ed, doesn't that mean that it's not possible that `originalParent` still having a strong reference to it?

How can `originalParent` be holding a reference to `this` in this code path?

Reply:
In general, a full graph of objects is collected at once, which means that the dependents are generally not collected yet. It may happen that the parent may still have been collected, but in this case, nothing will happen and the template will not be returned to the pool.

That being said, there's still a problem with the "activeinstances" list that I will take a look at (it's not cleared properly).